src/components/widgets/WeatherWidget/useGetWeatherData.ts

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useRef } from "react";
+import { Weather } from "../../../types";
+
+export const useGetWeatherData = (
+  setWeather: (weather: Weather) => void,
+  lat: number | null,
+  long: number | null
+) => {
+  const fetchWeatherData = () => {
+    if (lat === 0 && long === 0) {
+      return;
+    }
+    fetch(
+      `https://api.met.no/weatherapi/locationforecast/2.0/compact?lat=${lat}&lon=${long}`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setWeather({
+          windSpeed:
+            data.properties.timeseries[0].data.instant.details.wind_speed,
+          icon: data.properties.timeseries[0].data.next_1_hours.summary
+            .symbol_code,
+          airTemperature:
+            data.properties.timeseries[0].data.instant.details.air_temperature,
+          humidity:
+            data.properties.timeseries[0].data.instant.details
+              .relative_humidity,
+        });
+      });
+  };
+
+  const intervalRef = useRef(0);
+
+  useEffect(() => {
```

Comment:
Weather data should also be fetched on page load (component mount)

Reply:
As far as I can see the dependary array for this useEffect is empty, making it fetch on mount?

Reply:
Nah, it waits 5 minutes until it fetches.